package/src/components/ImageGallery/components/AnimatedGalleryVideo.tsx

```diff
@@ -0,0 +1,226 @@
+  videoRef: React.RefObject<VideoType>;
+  style?: StyleProp<ViewStyle>;
+};
+
+const styles = StyleSheet.create({
+  activityIndicator: {
+    alignSelf: 'center',
+  },
+  videoPlayer: {
+    height: '100%',
+    width: '100%',
+  },
+});
+
+export const AnimatedGalleryVideo: React.FC<Props> = React.memo(
+  (props) => {
+    const [opacity, setOpacity] = useState<number>(1);
+    const {
+      handleEnd,
+      handleLoad,
+      handleProgress,
+      index,
+      offsetScale,
+      paused,
+      previous,
+      scale,
+      screenHeight,
+      selected,
+      shouldRender,
+      source,
+      style,
+      translateX,
+      translateY,
+      videoRef,
+    } = props;
+
+    const onLoadStart = () => {
+      setOpacity(1);
+    };
+
+    const onLoad = (payload: VideoPayloadData) => {
+      setOpacity(0);
+      handleLoad(payload);
+    };
+
+    const onEnd = () => {
+      handleEnd();
+    };
+
+    const onProgress = (data: VideoProgressData) => {
+      handleProgress(data);
+    };
+
+    const onBuffer = ({ isBuffering }: { isBuffering: boolean }) => {
+      if (isBuffering) setOpacity(1);
+      else setOpacity(0);
+    };
+
+    const onPlayBackStatusUpdate = (playbackStatus: PlaybackStatus) => {
+      if (!playbackStatus.isLoaded) {
+        // Update your UI for the unloaded state
+        setOpacity(1);
+        if (playbackStatus.error) {
+          console.log(`Encountered a fatal error during playback: ${playbackStatus.error}`);
+        }
+      } else {
+        // Update your UI for the loaded state
+        setOpacity(0);
+        handleLoad({ duration: playbackStatus.durationMillis / 1000 });
+        if (playbackStatus.isPlaying) {
+          // Update your UI for the playing state
+          handleProgress({
+            currentTime: playbackStatus.positionMillis / 1000,
+            seekableDuration: playbackStatus.durationMillis / 1000,
+          });
+        } else {
+          // Update your UI for the paused state
+        }
+
+        if (playbackStatus.isBuffering) {
+          // Update your UI for the buffering state
+          setOpacity(1);
+        }
+
+        if (playbackStatus.didJustFinish && !playbackStatus.isLooping) {
+          // The player has just finished playing and will stop. Maybe you want to play something else?
+          handleEnd();
+        }
+      }
+    };
+
+    const animatedViewStyles = useAnimatedStyle<ViewStyle>(() => {
+      const xScaleOffset = -7 * screenWidth * (0.5 + index);
+      const yScaleOffset = -screenHeight * 3.5;
+      return {
+        transform: [
+          {
+            translateX: selected
+              ? translateX.value + xScaleOffset
+              : scale.value < 1 || scale.value !== offsetScale.value
+              ? xScaleOffset
+              : previous
+              ? translateX.value - halfScreenWidth * (scale.value - 1) + xScaleOffset
+              : translateX.value + halfScreenWidth * (scale.value - 1) + xScaleOffset,
```

Comment:
Can we refactor this please? It's hard to decipher what's going on here with a large nested ternary, and nothing to really tell us what the goal of the code is

Reply:
Kind of difficult to assign the variable names for the condition as even I could not decipher the logic yet. May be we can pair programme to get it done 😆